src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1710,6 +1710,48 @@ public async Task<bool> IsAsyncMethod(SessionId sessionId, int methodId, Cancell
             return retDebuggerCmdReader.ReadByte() == 1 ; //token
         }
 
+        public async Task<JArray> GetHoistedLocalVariables(SessionId sessionId, List<int> objectAlreadyRead, JArray asyncLocals, CancellationToken token)
+        {
+            JArray asyncLocalsFull = new JArray();
+            foreach (var asyncLocal in asyncLocals)
+            {
+                if (asyncLocal["name"].Value<string>().EndsWith("__this"))
+                {
+                    asyncLocal["name"] = "this";
+                    asyncLocalsFull.Add(asyncLocal);
+                }
+                else if (asyncLocal["name"].Value<string>().StartsWith("<>8__") ||
+                        asyncLocal["name"].Value<string>().StartsWith("<>f__ref") ||
+                        asyncLocal["name"].Value<string>().StartsWith("$locvar") ||
+                        asyncLocal["name"].Value<string>().StartsWith("CS$<>")) //same code that has on debugger-libs
+                {
+                    DotnetObjectId.TryParse(asyncLocal?["value"]?["objectId"]?.Value<string>(), out DotnetObjectId dotnetObjectId);
+                    int objectIdToGetInfo = int.Parse(dotnetObjectId.Value);
+                    if (!objectAlreadyRead.Contains(objectIdToGetInfo))
+                    {
+                        objectAlreadyRead.Add(objectIdToGetInfo);
+                        var asyncLocalsFromObject = await GetObjectValues(sessionId, objectIdToGetInfo, true, false, false, false, token);
+                        var hoistedLocalVariable = await GetHoistedLocalVariables(sessionId, objectAlreadyRead, asyncLocalsFromObject, token);
+                        asyncLocalsFull = new JArray(asyncLocalsFull.Union(hoistedLocalVariable));
+                    }
+                }
+                else if (asyncLocal["name"].Value<string>().Contains("<>"))
```

Comment:
Is the `<>` is a specific part of the string, or anywhere?

Reply:
changed to startswith.